src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -276,4 +277,44 @@ public ResponseEntity<Result> createResultForExternalSubmission(@PathVariable Lo
         return ResponseEntity.created(new URI("/api/results/" + savedResult.getId()))
                 .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, savedResult.getId().toString())).body(savedResult);
     }
+
+    /**
+     * GET exercises/:exerciseId/feedback-details : Retrieves all feedback details and the latest result IDs for a given exercise.
+     *
+     * @param exerciseId The ID of the exercise for which feedback details and result IDs should be retrieved.
+     * @return A ResponseEntity containing a list of all feedback details and the corresponding result IDs for the exercise.
+     */
+    @GetMapping("exercises/{exerciseId}/feedback-details")
+    @EnforceAtLeastTutor
+    public ResponseEntity<FeedbackDetailsResponse> getAllFeedbackDetailsForExercise(@PathVariable Long exerciseId) {
+        log.debug("REST request to get all Feedback details for Exercise {}", exerciseId);
+
+        Set<StudentParticipation> participations = resultRepository.findByExerciseIdWithLatestResultAndFeedbackElseThrow(exerciseId);
+        removeSubmissionAndExerciseData(participations);
+
+        List<Feedback> allFeedback = new ArrayList<>();
+        List<Long> resultIds = new ArrayList<>();
+
+        participations.forEach(participation -> {
+            participation.getResults().forEach(result -> {
+                resultIds.add(result.getId());
+                allFeedback.addAll(result.getFeedbacks());
+            });
+        });
+
+        FeedbackDetailsResponse response = new FeedbackDetailsResponse(allFeedback, resultIds);
+        return new ResponseEntity<>(response, HttpStatus.OK);
+    }
+
+    private void removeSubmissionAndExerciseData(Set<StudentParticipation> participations) {
+        // remove unnecessary data to reduce response size
+        participations.forEach(participation -> {
+            participation.setSubmissions(null);
+            participation.setExercise(null);
+        });
+    }
+
+    public record FeedbackDetailsResponse(List<Feedback> feedback, List<Long> resultIds) {
```

Comment:
Can you move this into its own class in the `dto` sub package and also add `DTO` to the end of its name?
And new DTOs should not contain domain objects like `Feedback`